src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java

```diff
@@ -52,6 +52,15 @@
   @VisibleForTesting static final String PARAM_TIMESTAMP = "timestamp";
   @VisibleForTesting static final String PARAM_TYPE = "type";
 
+  @VisibleForTesting static final String ERROR_MISSING_LECTURE = "Missing lecture parameter.";
+  @VisibleForTesting
+  static final String ERROR_MISSING_COMMENT_TYPE = "Missing comment type parameter.";
+  @VisibleForTesting
+  static final String ERROR_MISSING_PARENT = "Missing parent parameter for reply comment.";
+  @VisibleForTesting
+  static final String ERROR_MISSING_TIMESTAMP = "Missing timestamp parameter for root comment.";
+  @VisibleForTesting static final String ERROR_NOT_LOGGED_IN = "You are not logged in.";
```

Comment:
Concerning @VisibleForTesting, for static fields it makes more sense to have these as public and call them directly in tests. In general @VisibleForTesting should be last resort after considering other options.

Reply:
Should PARAMs be public too?  I thought we wanted to make them the most protected as possible, so used @VisibleForTesting because they can't be private

Reply:
"If you expose constants in Java, please mark the modifications with @VisibleForTesting"
This is from the doc: https://docs.google.com/document/d/18T9ef-DRqTncVD377GoIQmOGI84cNKkZ7Nn07YURFTw/edit

Reply:
https://docs.google.com/document/d/1ogapKfwinMeWHbJxMHzjrGxpoAiz-r736CXQ8y5CPxY/edit#heading=h.fmos5zqp4q3d

I think the use of this is really a balance. A lot of the time it signifies a code smell if one needs to do this.
Maybe the right thing here is to move this strings to a Util  file like we had for the other Servlets. If many strings now need to be shared outside of this class they should be in a DiscussionUtil class under /utils like the other Servlets.

Reply:
I don't think making visible for testing is so bad. This is only exposing a variable, not a method. I don't think it makes sense to move it to a Util because they are only used in this class. If you do not want to make @VisibleForTesting, then when testing the error message just pass the string directly in. It is not common practice to create const variables for error messages. Just do:

assertThat(response.errorMessage).isEqualTo("error message").

Reply:
Spoke to @billie-cat  offline. The  @VisibleForTesting usage is truly a balance and if there are too many in a file its definitely a code smell. I'm ok with them being  inline in the assertions.

Please remove @VisbleForTesting and check inline like @billie-cat has mentioned.
```assertThat(response.errorMessage).isEqualTo("error message")```.

In general if a string is being used in one place in a test lets start making those inline to prevent this confusion.

Reply:
Done in all 3 PRs!  Thanks for figuring out the best way to do this :)